src/ClientMessage.elm

```diff
@@ -81,3 +96,12 @@ encodeLocation loc =
         , ( "username", Encode.string loc.username )
         , ( "password", Encode.string loc.password )
         ]
+
+
+
+-- Publish
+
+
+publishDecoder : Decoder ClientMessage
+publishDecoder =
+    Decode.map Publish CommonMessages.publicationDecoder
```

Comment:
For folks not familiar with decoders, this is describing the process of "Decode data with `CommonMessages.publicationDecoder` (which decodes to Publication) and then use the `Publish` constructor to tag it as a variant of `ClientMessage`"